shell/components/form/NodeScheduling.vue

```diff
@@ -127,7 +127,33 @@ export default {
       }
     },
     isEmpty
-  }
+  },
+
+  watch: {
+    'value.nodeSelector': {
+      handler(nodeSelector) {
+        if (this.isHarvester && nodeSelector?.[HOSTNAME]) {
+          this.selectNode = 'nodeSelector';
+          const nodeName = nodeSelector[HOSTNAME];
+
+          this.nodeName = nodeName;
+
+          const array = this.nodes.map(n => n.value);
+
+          if (nodeName && !array.includes(nodeName)) {
+            this.$store.dispatch('growl/error', {
+              title:   this.$store.getters['i18n/t']('harvester.vmTemplate.tips.notExistNode.title', { name: nodeName }),
+              message: this.$store.getters['i18n/t']('harvester.vmTemplate.tips.notExistNode.message')
+            }, { root: true });
+          }
+
+          delete this.value.nodeSelector;
+          this.$set(this, 'nodeName', '');
+          this.$set(this, 'selectNode', null);
```

Comment:
why delete?

Data is cleared when saved.

Reply:
It also needs to empty when viewing the detail page. same as https://github.com/harvester/dashboard/blob/a0e953e9348d2ba271184428de605a08db5077e1/shell/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/type/vmImage.vue#L171